middleware/app/Http/Controllers/ProviderPlatformController.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use ProviderServices;
+
+class ProviderPlatformController extends Controller
+{
+    // ****************************************************
+    //  POST: /api/provider_platform/{request_body}
+    // Request $req example:
+    //
+    // { "type": "canvas", "account_id": 123, "account_name": "WashU",
+    // "access_key": "12u3n4gh3k69jj21k_gh27", "base_url": "https://canvas.instructure.com",
+    // "iconUrl": "https://canvas.instructure.com/favicon.ico", "comsumer_id": 1 }
+    //
+    // ****************************************************
+    public function registerProviderConnection(Request $req): \Illuminate\Http\JsonResponse
```

Comment:
Let's only have a CRUD interface to provider platforms here. We can then have calls in PlatformConnectionController for mapping consumer platforms and provider platforms.